src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -2733,10 +2735,100 @@ namespace winrt::Microsoft::Terminal::Control::implementation
         }
     }
 
+    winrt::fire_and_forget TermControl::_updateSelectionMarkers(IInspectable /*sender*/, Control::UpdateSelectionMarkersEventArgs args)
+    {
+        auto weakThis{ get_weak() };
+        co_await resume_foreground(Dispatcher());
+        if (weakThis.get() && args)
+        {
+            if (_core.HasSelection() && !args.ClearMarkers())
+            {
+                // lambda helper function that can be used to display a selection marker
+                // - targetEnd: if true, target the "end" selection marker. Otherwise, target "start".
+                auto displayMarker = [&](bool targetEnd) {
+                    // Compute the location of the top left corner of the cell in DIPS
+                    const auto terminalPos{ targetEnd ? _core.SelectionEnd() : _core.SelectionAnchor() };
+                    const til::point locationInDIPs{ _toPosInDips(terminalPos) };
+
+                    // Move the marker to the top left corner of the cell
+                    const auto& marker{ targetEnd ? SelectionEndMarker() : SelectionStartMarker() };

```

Comment:
should we just pass the marker in as the param?

Reply:
.... no cause we need `targetEnd` to figure out where to draw the marker. Okay, that's fine. Just looked like duplicated code. Ignore me.